src/tests.ts

```diff
@@ -146,53 +184,57 @@ async function allowOrder(arweave, walletA, contractA, contractB) {
   await arweave.transactions.sign(allowTx, walletA.jwk);
   await arweave.transactions.post(allowTx);
 
-  await mine();
+  await mine();*/
 
-  return allowTx.id;
+  return originalTxId;
 }
 
-async function makeOrder(arweave, walletA, contractA, contractB, allowTx) {
-  const orderTx = await arweave.createTransaction(
-    {
-      data: "1234",
-    },
-    walletA.jwk
-  );
-  const input = {
+async function makeOrder(
+  walletA: JWKInterface,
+  contractA: Contract<StateInterface>,
+  contractB: Contract<StateInterface>,
+  allowTx: string
+) {
+  let contract = allowTx === "" ? contractA : contractB;
+
+  const { originalTxId } = await contract.connect(walletA).writeInteraction({
     function: "createOrder",
     transaction: allowTx,
-    pair: [contractA, contractB],
+    pair: [contractA.txId(), contractB.txId()],
     qty: 10,
     price: 1,
-  };
-
-  orderTx.addTag("App-Name", "SmartWeaveAction");
-  orderTx.addTag("App-Version", "0.3.0");
-  if (allowTx === "") {
-    // Order is on itself
-    orderTx.addTag("Contract", contractA);
-  } else {
-    orderTx.addTag("Contract", contractB);
-  }
-  orderTx.addTag("Input", JSON.stringify(input));
-
-  await arweave.transactions.sign(orderTx, walletA.jwk);
-  await arweave.transactions.post(orderTx);
```

Comment:
btw...if internal writes are being used, the transactions MUST be posted via contract.writeInteraction - as it does some additional work:
1. makes a dry-write
2. analyzes the call tree and verifies if there's a internal write
3. if so - generates additional tag 

It is described here https://github.com/warp-contracts/warp/issues/37 (Detailed description -> Solution -> point 2)